src/utils/Once.js

```diff
@@ -0,0 +1,17 @@
+class Once {
+  constructor(f) {
+    this.oncePromise = f().then(() => {
+      this.done = true;
+    }).catch(error => {
+      console.error(`Fatal exception caught: ${error}`);
+      throw error;
+    });
+    this.done = false;
```

Comment:
I feel this is a bit weird, f().then... executes immediately and you set this.done = true in the arrow function for then and then set this.done = false right before returning for constructor.  Do I have that right?

Reply:
Yeah you're right.  This should be set to false earlier on the off chance that the promise returns synchronously.